### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by Stallind**
Added homework 4

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Globalization;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734&bid=16182861)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734&bid=16182861)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734&bid=16182861)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734&bid=16182861)

**Comment by Stallind** [comment]
> Your solution looks good!
> One thing that I have noticed is that you have mixed \n and Environment.NewLine. This is bad in my opinion because it can make troubleshooting way way way harder than it has to be.
> In order to fix this you can simply replace all occurences.
> 
> In order to keep your output text readable you may want to define a
> `private static string newline = Environment.NewLine;`
> variable at class level scope or create a private function like
> `private static void WriteNewLines(int count)`.
> 
> Using the function is preferable because it is much more maintainable. :)

I completly agree, fixed

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            }
+        }
+
+        private static string WriteNewLine(int count)
+        {
+            string newLines = Environment.NewLine;
+
+            for (int i = 1; i < count; i++)
+            {
+                newLines += Environment.NewLine;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734&bid=16182861)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4884734&bid=16182861)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    public class Lesson3
+    {
+        public static void Demo()
+        {
+            const int maxEntries = 2;
+
+            for (int i = 1; i <= maxEntries; i++)
+            {
+                Console.Clear();
+
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: A blank line in between every line of code is not needed.

Separate unrelated code with a blank line and if all code is unrelated- such separation is not needed.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                newLines.Append(Environment.NewLine);
+            }
+
+            return newLines.ToString();
+        }
+
+        static void RegisterNewPersonData(int entry)
+        {
+            Console.WriteLine($"Enter the following information for person #{entry}.");
+
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: This is a good example of separating stuff with an empty line. You actually separated concepts.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            string firstName = RegisterStringValue("Firstname: ");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Could be done in a single prompt because name and surname are not treated differently.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        public static void Summarize(string firstName, string lastName, int age, float weight, float height, float bmi)
+        {
+            Console.WriteLine($"{WriteNewLine(1)}{firstName} {lastName} is {age} years old and his weight is {weight} and his height is {height} \n \n");
+
+            Console.WriteLine($"He/she has a BMI of {bmi} {WriteNewLine(3)}");
+        }
+
+        public static void PromptNewPerson(int entry)
+        {
+            Console.WriteLine(entry < 2
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Magic number. Please use named constant to avoid magic numbers.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            if (height <= 0 && weight <= 0)
+            {
+                Console.WriteLine($"Failed calculating BMI. Reason:");
+                Console.WriteLine($"Weight cannot be equal or less than zero, but was {weight}.");
+                Console.WriteLine($"Height cannot be less than zero, but was {height}.");
+
+                return -1;
+            }
+
+            if (height <= 0 || weight <= 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Could also go to a separate function. Such as ValidateBody()
